packages/api/tests/task.test.js

```diff
@@ -2339,4 +2349,53 @@ describe('Task tests', () => {
       });
     });
   });
+
+  describe('Patch task wage test', () => {
+    const testWithAdminRoleAndValidInput = async (userRoleId, wage_at_moment, done) => {
+      const patchTaskWageBody = { wage_at_moment };
+      const [{ user_id, farm_id }] = await mocks.userFarmFactory({}, fakeUserFarm(userRoleId));
+      const [{ task_id }] = await mocks.taskFactory({ promisedUser: [{ user_id }] });
+      const [{ location_id }] = await mocks.locationFactory({ promisedFarm: [{ farm_id }] });
+      await mocks.location_tasksFactory({
+        promisedTask: [{ task_id }],
+        promisedField: [{ location_id }],
+      });
+
+      patchTaskWageRequest({ user_id, farm_id }, patchTaskWageBody, task_id, async (err, res) => {
+        expect(res.status).toBe(200);
+        const updated_task = await getTask(task_id);
+        expect(updated_task.wage_at_moment).toBe(wage_at_moment);
+        done();
+      });
+    };
+
+    test('Farm owner must be able to patch task wage', async (done) => {
+      testWithAdminRoleAndValidInput(1, 33, done);
+    });
+
+    test('EO must be able to patch task wage', async (done) => {
+      testWithAdminRoleAndValidInput(5, 27, done);
+    });
+
+    test('Managers must be able to patch task wage', async (done) => {
+      testWithAdminRoleAndValidInput(2, 37, done);
+    });
+
+    test('Farm worker must not be able to patch task due date', async (done) => {
```

Comment:
Test name should be wage instead of task due date. And just wondering if you could reuse the same function testAdminRoleWithValidInput() here by splitting the expect statement by roleId.

Reply:
Good catch, thank you! I agree, I think reusing the function is better.